contracts/src/L2StandardBridgeBot.sol

```diff
@@ -66,12 +68,30 @@ contract L2StandardBridgeBot is Ownable {
         withdrawTo(_l2Token, msg.sender, _amount, _minGasLimit, _extraData);
     }
 
-    // withdrawFee withdraw the delegation fee vault to _recipient address, only owner can call this function.
+    // withdrawFee withdraw the delegation fee vault to _recipient address on L2, only owner can call this function.
     function withdrawFee(address _recipient) external onlyOwner {
         (bool sent, ) = _recipient.call{ value: address(this).balance }("");
         require(sent, "Failed to send Ether");
     }
 
+    // withdrawFeeToL1 withdraw the delegation fee vault to _recipient address on L1, only owner can call this function.
+    //
+    // NOTE: Since the `withdrawTo` function is defined as external and `_extraData` is `bytes calldata`, the
+    // `_extraData` cannot be mock from `bytes memory` by the caller. See also https://github.com/ethereum/solidity/issues/12778
+    function withdrawFeeToL1(address _recipient, bytes calldata _extraData) external onlyOwner {
+        uint256 _balance = address(this).balance;
+        require(_balance > delegationFee, "fee vault balance is insufficient to pay the required amount");
+
+        uint256 _amount = _balance - delegationFee;
+        this.withdrawTo{ value: _balance }(
```

Comment:
`this.` should be redundant since `withdrawTo` is a public function

Reply:
This call change the msg.value via `{ value: _balance }`, and only external calls are allow to do it.

See also "Error (2193): Function call options can only be set on external function calls or contract creations."

Reply:
I believe you can remove `{ value: _balance }` since this is an internal call, the context of `msg.value` will be passed.
Check the suggestion below.
And plz add a test for this function to confirm.

Reply:
WithdrawFeeToL1 is intended to be:

1. Owner sends the transaction, with `tx.value = 0`

2. `withdrawFeeToL1` transfers all the balance of `L2StandardBridgeBot` contract to `_recipient` by specifying `this.withdrawTo{ value: _balance }`

Without `{ value: _balance }`, the call to `L2StandardBridge.withdrawTo` will be `msg.value = 0`, that is not my intention.

Reply:
Enhance the test case at [798a563](https://github.com/bnb-chain/opbnb-bridge-bot/pull/9/commits/798a563c1eebd4076cde72159504fbf3e00f9492), PTAL

Reply:
I see